src/main/java/org/broadinstitute/hellbender/utils/fragments/FragmentCollection.java

```diff
@@ -0,0 +1,144 @@
+package org.broadinstitute.hellbender.utils.fragments;
+
+import org.broadinstitute.hellbender.utils.pileup.PileupElement;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.*;
+import java.util.function.Function;
+
+/**
+ * Represents the results of the reads -> fragment calculation.
+ *
+ * Contains singleton -- objects whose underlying reads do not overlap their mate pair
+ * Contains overlappingPairs -- objects whose underlying reads do overlap their mate pair
+ */
+public final class FragmentCollection<T> {
```

Comment:
This class lacks units tests -- at the very least, we need tests for `create()`, as it's non-trivial